Amazon.IonDotnet.Tests/Internals/ReaderLocalTableTest.cs

```diff
@@ -13,40 +13,299 @@
+                Assert.AreEqual("s1", binaryRoundTrip.GetElementAt(1).StringValue);
+                Assert.AreEqual("s2", binaryRoundTrip.GetElementAt(2).StringValue);
+                Assert.AreEqual("s3", binaryRoundTrip.GetElementAt(3).StringValue);
+            }
+        }
+
+        [TestMethod]
+        public void TestLocalSymbolTableMultiAppend()
+        {
+            var text =
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  symbols:[ \"s1\", \"s2\" ]" +
+                    "}\n" +
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  imports:" + SystemSymbols.IonSymbolTable + "," +
+                    "  symbols:[ \"s3\" ]" +
+                    "}\n" +
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  imports:" + SystemSymbols.IonSymbolTable + "," +
+                    "  symbols:[ \"s4\", \"s5\" ]" +
+                    "}\n" +
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  imports:" + SystemSymbols.IonSymbolTable + "," +
+                    "  symbols:[ \"s6\" ]" +
+                    "}\n" +
+                    "null";
+
+            ISymbolTable symbolTable = OneValue(text);
+            CheckLocalTable(symbolTable);
+
+            var systemMaxId = symbolTable.GetSystemTable().MaxId;
+
+            // Table contains all symbols and SIDs are in correct order.
+            CheckSymbol("s1", systemMaxId + 1, symbolTable);
+            CheckSymbol("s2", systemMaxId + 2, symbolTable);
+            CheckSymbol("s3", systemMaxId + 3, symbolTable);
+            CheckSymbol("s4", systemMaxId + 4, symbolTable);
+            CheckSymbol("s5", systemMaxId + 5, symbolTable);
+            CheckSymbol("s6", systemMaxId + 6, symbolTable);
+
+            CheckUnknownSymbol("unknown", SymbolToken.UnknownSid, symbolTable);
+            CheckUnknownSymbol(33, symbolTable);
+        }
+
+        [TestMethod]
+        public void TestLocalSymbolTableAppendEmptyList()
+        {
+            var original =
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  symbols:[ \"s1\" ]" +
+                    "}\n";
+
+            var appended = original +
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  imports:" + SystemSymbols.IonSymbolTable + "," +
+                    "  symbols:[]" +
+                    "}\n";
+
+            ISymbolTable originalSymbolTable = OneValue(original + "null");
+            ISymbolTable appendedSymbolTable = OneValue(appended + "null") ;
+
+            var originalSymbol = originalSymbolTable.Find("s1");
+            var appendedSymbol = appendedSymbolTable.Find("s1");
+
+            Assert.AreEqual(originalSymbol.Sid, appendedSymbol.Sid);
+        }
+
+        private static ISymbolTable OneValue(string text)
```

Comment:
A more suitable name for this method seems to be something like `ReadSymbolTable`.

Reply:
Done.